interface/procedure_tools/quest/gen_hl7_order.inc.php

```diff
@@ -0,0 +1,590 @@
+              $datatype = 'ST';
+            if ($fldtype == 'N') {
+                $datatype = "NM";
+            } else if ($fldtype == 'D') {
+                  $answer = hl7Date($answer);
+            } else if ($fldtype == 'G') {
+                  $weeks = intval($answer / 7);
+                  $days = $answer % 7;
+                  $answer = $weeks . 'wks ' . $days . 'days';
+            }
+
+              $out .= "OBX" .
+                $d1 . ++$setid2 .                           // Set ID
+                $d1 . $datatype .                           // Structure of observation value
+                $d1 . hl7Text($qrow['question_code']) .     // Clinical question code
+                $d1 .
+                $d1 . hl7Text($answer) .                    // Clinical question answer
+                $d0;
+        }
+    }
+        //recreate OBX here to add to end of hl7
+        $out .= "OBX" .
+            $d1 . "1" .
+            $d1 . "ST" .
+            $d1 . "PR101^CLINICAL INFORMATION" .
+            $d1 .
+            $d1 . $porow['patient_instructions'] .
+            $d0;
+
+        $out .= "OBX" .
+            $d1 . "2" .
+            $d1 . "ST" .
+            $d1 . "PR300^SPECIMEN SOURCE/SAMPLE TYPE" .
+            $d1 .
+            $d1 . $porow['clinical_hx'] .
+            $d0;
+
+    return '';
+}
+
+/**
+ * Transmit HL7 for the specified lab.
+ *
+ * @param  integer $ppid  Procedure provider ID.
+ * @param  string  $out   The HL7 text to be sent.
+ * @return string         Error text, or empty if no errors.
+ */
+function send_hl7_order($ppid, $out)
+{
+    global $srcdir;
+
+    $d0 = "\r";
+
+    $pprow = sqlQuery("SELECT * FROM procedure_providers " .
+    "WHERE ppid = ?", array($ppid));
+    if (empty($pprow)) {
+        return xl('Procedure provider') . " $ppid " . xl('not found');
+    }
+
+    $protocol = $pprow['protocol'];
+    $remote_host = $pprow['remote_host'];
+
+  // Extract MSH-10 which is the message control ID.
+    $segmsh = explode(substr($out, 3, 1), substr($out, 0, strpos($out, $d0)));
+    $msgid = $segmsh[9];
+    if (empty($msgid)) {
+        return xl('Internal error: Cannot find MSH-10');
+    }
+
+    if ($protocol == 'DL' || $pprow['orders_path'] === '') {
+        header("Pragma: public");
+        header("Expires: 0");
+        header("Cache-Control: must-revalidate, post-check=0, pre-check=0");
+        header("Content-Type: application/force-download");
+        header("Content-Disposition: attachment; filename=order_$msgid.hl7");
+        header("Content-Description: File Transfer");
+        echo $out;
+        exit;
+    } else if ($protocol == 'SFTP') {
+        // Compute the target path/file name.
+        $filename = $msgid . '.txt';
+        if ($pprow['orders_path']) {
+            $filename = $pprow['orders_path'] . '/' . $filename;
+        }
+
+        // Connect to the server and write the file.
+        $sftp = new \phpseclib\Net\SFTP($remote_host);
```

Comment:
note most recent codebase is now using phpseclib3 instead of phpseclib (just need to add the 3 above and it should work fine)